sql/core/src/test/resources/sql-tests/inputs/pgSQL/case.sql

```diff
@@ -0,0 +1,267 @@
+
+-- Test for cases involving untyped literals in test expression
+SELECT CASE 'a' WHEN 'a' THEN 1 ELSE 2 END;
+
+--
+-- Examples of targets involving tables
+--
+
+SELECT '' AS `Five`,
+  CASE
+    WHEN i >= 3 THEN i
+  END AS `>= 3 or Null`
+  FROM CASE_TBL;
+
+SELECT '' AS `Five`,
+  CASE WHEN i >= 3 THEN (i + i)
+       ELSE i
+  END AS `Simplest Math`
+  FROM CASE_TBL;
+
+SELECT '' AS `Five`, i AS `Value`,
+  CASE WHEN (i < 0) THEN 'small'
+       WHEN (i = 0) THEN 'zero'
+       WHEN (i = 1) THEN 'one'
+       WHEN (i = 2) THEN 'two'
+       ELSE 'big'
+  END AS `Category`
+  FROM CASE_TBL;
+
+SELECT '' AS `Five`,
+  CASE WHEN ((i < 0) or (i < 0)) THEN 'small'
+       WHEN ((i = 0) or (i = 0)) THEN 'zero'
+       WHEN ((i = 1) or (i = 1)) THEN 'one'
+       WHEN ((i = 2) or (i = 2)) THEN 'two'
+       ELSE 'big'
+  END AS `Category`
+  FROM CASE_TBL;
+
+--
+-- Examples of qualifications involving tables
+--
+
+--
+-- NULLIF() and COALESCE()
+-- Shorthand forms for typical CASE constructs
+--  defined in the SQL standard.
+--
+
+SELECT * FROM CASE_TBL WHERE COALESCE(f,i) = 4;
+
+SELECT * FROM CASE_TBL WHERE NULLIF(f,i) = 2;
+
+SELECT COALESCE(a.f, b.i, b.j)
+  FROM CASE_TBL a, CASE2_TBL b;
+
+SELECT *
+  FROM CASE_TBL a, CASE2_TBL b
+  WHERE COALESCE(a.f, b.i, b.j) = 2;
+
+SELECT '' AS Five, NULLIF(a.i,b.i) AS `NULLIF(a.i,b.i)`,
+  NULLIF(b.i, 4) AS `NULLIF(b.i,4)`
+  FROM CASE_TBL a, CASE2_TBL b;
+
+SELECT '' AS `Two`, *
+  FROM CASE_TBL a, CASE2_TBL b
+  WHERE COALESCE(f,b.i) = 2;
+
+-- We don't support update now.
+--
+-- Examples of updates involving tables
+--
+
+-- UPDATE CASE_TBL
+--   SET i = CASE WHEN i >= 3 THEN (- i)
+--                 ELSE (2 * i) END;
+
+-- SELECT * FROM CASE_TBL;
+
+-- UPDATE CASE_TBL
+--   SET i = CASE WHEN i >= 2 THEN (2 * i)
+--                 ELSE (3 * i) END;
+
+-- SELECT * FROM CASE_TBL;
+
+-- UPDATE CASE_TBL
+--   SET i = CASE WHEN b.i >= 2 THEN (2 * j)
+--                 ELSE (3 * j) END
+--   FROM CASE2_TBL b
+--   WHERE j = -CASE_TBL.i;
+
+-- SELECT * FROM CASE_TBL;
+
+-- We don't support the features below:
+-- 1. CREATE FUNCTION ... returns ... as ...
+-- 2. CREATE DOMAIN ...
+-- 3. CREATE OPERATOR ...
+-- 4. CREATE TYPE ...
+--
+-- Nested CASE expressions
+--
+
+-- This test exercises a bug caused by aliasing econtext->caseValue_isNull
+-- with the isNull argument of the inner CASE's CaseExpr evaluation.  After
+-- evaluating the vol(null) expression in the inner CASE's second WHEN-clause,
+-- the isNull flag for the case test value incorrectly became true, causing
+-- the third WHEN-clause not to match.  The volatile function calls are needed
+-- to prevent constant-folding in the planner, which would hide the bug.
+
+-- Wrap this in a single transaction so the transient '=' operator doesn't
+-- cause problems in concurrent sessions
+-- BEGIN;
+
+-- CREATE FUNCTION vol(text) returns text as
+--   'begin return $1; end' language plpgsql volatile;
+
+-- SELECT CASE
+--   (CASE vol('bar')
```

Comment:
This test case is pretty useful. Could we use a udf here?

Reply:
Yes. Add `vol` to  `SQLQueryTestSuite`.